processor/src/chiplets/mod.rs

```diff
@@ -89,38 +95,95 @@ impl AuxTable {
         self.hasher.trace_len() + self.bitwise.trace_len()
     }
 
-    /// Adds all range checks required by the memory co-processor to the provided [RangeChecker]
+    // LOOKUP ACCESSORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns a word (4 elements) located in memory at the specified address.
+    ///
+    /// If the specified address hasn't been previously written to, four ZERO elements are
+    /// returned. This effectively implies that memory is initialized to ZERO.
+    pub fn readw_mem(&mut self, addr: Felt) -> Word {
+        // read the word from memory
+        let value = self.memory.read(addr);
+
+        // send the memory read request to the bus
+        self.bus
+            .request_memory_operation(addr, self.clk, value, value);
+
+        value
+    }
+
+    /// Writes the provided element to memory at the specified address leaving the remaining 3
+    /// elements of the word previously stored at that address unchanged.
+    pub fn write_mem(&mut self, addr: Felt, value: Felt) -> Word {
+        let old_word = self.memory.get_old_value(addr);
+        let word = [value, old_word[1], old_word[2], old_word[3]];
+
+        self.memory.write(addr, word);
+
+        // send the memory write request to the bus
+        self.bus
+            .request_memory_operation(addr, self.clk, old_word, word);
+
+        old_word
+    }
+
+    /// Writes the provided word (4 elements) to memory at the specified address.
+    pub fn writew_mem(&mut self, addr: Felt, word: Word) -> Word {
+        let old_word = self.memory.get_old_value(addr);
+        self.memory.write(addr, word);
+
+        // send the memory write request to the bus
+        self.bus
+            .request_memory_operation(addr, self.clk, old_word, word);
+
+        old_word
+    }
```

Comment:
nit: I wonder if these are too close. If we do want to have 2 separate methods, I'd probably go with something like this:

* `write_mem()` -> `write_mem_single()`.
* `writew_mem()` -> `write_mem()`.